src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -431,34 +432,40 @@ internal abstract partial class SafeDeleteContext : SafeHandle
             SafeFreeContextBuffer outFreeContextBuffer = null;
             try
             {
-                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[inSecurityBufferDescriptor.cBuffers];
-                inUnmanagedBuffer.Clear();
+                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[3];
+                for (int index = 0; index < inSecBuffers.Count; ++index)
+                {
+                    inUnmanagedBuffer[index].BufferType = inSecBuffers.GetBuffer(index).Type;
+                    inUnmanagedBuffer[index].cbBuffer = inSecBuffers.GetBuffer(index).Token.Length;
+                }
 
                 fixed (void* inUnmanagedBufferPtr = inUnmanagedBuffer)
-                fixed (void* pinnedToken0 = inSecBuffers.Length > 0 ? inSecBuffers[0].token : null)
-                fixed (void* pinnedToken1 = inSecBuffers.Length > 1 ? inSecBuffers[1].token : null)
-                fixed (void* pinnedToken2 = inSecBuffers.Length > 2 ? inSecBuffers[2].token : null) // pin all buffers, even if null or not used, to avoid needing to allocate GCHandles
+                fixed (void* pinnedToken0 = inSecBuffers.GetBuffer(0).Token)
+                fixed (void* pinnedToken1 = inSecBuffers.GetBuffer(1).Token)
+                fixed (void* pinnedToken2 = inSecBuffers.GetBuffer(2).Token)
```

Comment:
Previously when inSecBuffers was a span, the indexer was returning the entry by ref... now with GetBuffer, it's returning it by value.  If not a functional difference, there could be a perf impact.

Reply:
I saw your comment on the previous PR. I plan to run a benchmark on both variants next week. But I did not want to hold back the split of #1949 so progress can be made independently.

Reply:
Changing the return type of `GetBuffer` to `readonly ref` would definitely have (slightly) better performance.

Reply:
I could not get ref to the items so I removed GetBuffer() completely to avoid copy.